### DISCUSSION TITLE
How can simplify this rewrites config

### DISCUSSION THREAD

**Discussion by wildfrontend** [vercel/next.js#50984] [Help] [answered]
### Summary

i just no idea how to simplify  this sample , could any help

```
  rewrites: async () => {
    return [
      {
        source: '/:locale',
        destination: '/:locale/all-tab/',
      },
      {
        source: '/:locale/articles/:path*',
        destination: '/:locale/all-tab/articles/:path*',
      },
      {
        source: '/:locale/matches/:path*',
        destination: '/:locale/all-tab/matches/:path*',
      },
      {
        source: '/:locale/rankings/:path*',
        destination: '/:locale/all-tab/rankings/:path*',
      },
    ];
```

**Comment by devjiwonchoi** [comment]
```js
const paths = ['', '/articles', '/matches', '/rankings']

rewrites: async () => {
  return paths.map((el) => ({
    source: `/:locale${el}/:path*`,
    destination: `/:locale/all-tab${el}/:path*`,
  }))
}
```
This will do the work.

Also, is there a reason why you are not following the format in the example of Next.js App Router [`rewrites`](https://nextjs.org/docs/app/api-reference/next-config-js/rewrites)?

```js
// next.config.js

const paths = ['', '/articles', '/matches', '/rankings']

module.exports = {
  async rewrites() {
    return paths.map((el) => ({
      source: `/:locale${el}/:path*`,
      destination: `/:locale/all-tab${el}/:path*`,
    }))
  },
}

```

**Reply by wildfrontend** [reply]
at first , i try to use regexp to simplify, but i not not familiar enough how to set path regexp , i did not think of can also use array map

**Reply by devjiwonchoi** [reply, marked as answer]
Note that there could be various options and mine is just one of them. 

Regex is used to filter out the matches, but rewrite needs a `destination` corresponding to the `source`,
therefore even though you use the regex, you'll still need to declare the `source` and `destination`.

My answer could be helpful since all you need to do is change the path in the `paths` variable, and the rest will be automatically modified.

Example:

```js
// Modify here, and the paths will Add/Change/Remove automatically.
const paths = ['', '/articles', '/matches', '/rankings']

module.exports = {
  async rewrites() {
    return paths.map((el) => ({
      source: `/:locale${el}/:path*`,
      destination: `/:locale/all-tab${el}/:path*`,
    }))
  },
}
```
